src/components/UI/WorkCard/WorkCard.tsx

```diff
@@ -0,0 +1,57 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import { useTranslation } from "react-i18next";
+import JobTitle, {
+  Card,
+  JobDescription,
+} from "components/UI/WorkCard/WorkCardStyles";
+import dayjs from "dayjs";
+import { useState } from "react";
+import { useGetOneJobPostQuery } from "services/jobPosts/setJobPostsAPI";
+import SearchWorkDrawer from "components/UI/drawers/SearchWorkDrawer/SearchWorkDrawer";
+
+export interface IWorkCardProps {
+  id: number;
+  title: string;
+  createdAt: string;
+  jobDescription: string;
+}
+
+const wordPerCard = 30;
+
+function WorkCard({ jobDescription, title, createdAt, id }: IWorkCardProps) {
+  const { t } = useTranslation();
+
+  const [open, setOpen] = useState(false);
+  const { data, isSuccess } = useGetOneJobPostQuery({
+    id,
+  });
+
+  return (
+    <>
+      <Card
+        onClick={() => {
+          setOpen(true);
+        }}
+      >
+        <JobTitle font_sz="1.5em">
+          {title}
+          <JobTitle font_sz="0.7em">
+            {createdAt ? dayjs(createdAt).fromNow() : t("NotFound.notFound")}
+          </JobTitle>
+        </JobTitle>
+        <JobDescription>
+          {jobDescription.split(" ").slice(0, wordPerCard).join(" ")}...
```

Comment:
move this magic to utils, with function name that will describe what are you trying to do